x-pack/plugins/infra/public/observability_logs/log_stream_query_state/src/types.ts

```diff
@@ -36,38 +37,78 @@ export interface LogStreamQueryContextWithValidationError {
   validationError: Error;
 }
 
+export type ExtendedTimeRange = TimeRange & { lastChangedCompletely: number };
```

Comment:
`lastChangedCompletely` is used deeper in the hierarchy by entries as a "time to fetch" signal. I haven't looked to remove it in this PR, but we should evaluate it's need when we refactor Log Entries.